csv/write.go

```diff
@@ -58,6 +60,7 @@ func convertSchema(sch *arrow.Schema) *arrow.Schema {
 		if !isTypeSupported(f.Type) {
 			fields[i].Type = arrow.BinaryTypes.String
 		}
+		fields[i].Metadata = stripCQExtensionMetadata(fields[i].Metadata)
```

Comment:
This call is required for the csv writer to actually write the records, on write it compares the schema given to the `csv.NewWriter` with the record schema, which (in our case) turns out to be slightly different and missing the `cq:extension:*` things. So we strip them here from both places (since they don't hold any actionable info for csv writing)